### PR TITLE
Fixed bad Sphinx Needs test link

### PR DESCRIPTION
**Pull request by john-science**
## Description

The docs are currently broken in ARMI.

[The PR #1171](https://github.com/terrapower/armi/pull/1171) tried to duplicate a Sphinx Needs / requirement link in our docs.  (FYI @Ashlita6 )

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Review by opotowsky — APPROVED** [review]

File: doc/release/0.2.rst [hunk 1]
```diff
@@ -8,7 +8,9 @@ Release Date: TBD
 
 What's new in ARMI
 ------------------
-#. TBD
+#. TBD: Composites can now be nested inside other Composites
+#. TBD: The global CaseSettings object has been removed
+#. The method ``Material.density3`` is now called ``density``, and the old ``density`` is now called ``pseudoDensity``. (`PR#1163 <https://github.com/terrapower/armi/pull/1163>`_)
```
Comment (keckler) [inline doc/release/0.2.rst:13]: These changes are unrelated to this PR? Do you need to merge main?

**Review by keckler — CHANGES_REQUESTED** [review]

File: doc/release/0.2.rst (hunk 1 above)
Comment (john-science) [reply doc/release/0.2.rst:13]: Unrelated to this PR. Sure, I can move them to their own PR.

I just want to release ARMI version `0.3.0` as soon as these three topics are done (though that third one is probably more like "Materials were reworked to be more consistent").

So I will want to put those three TBDs at the top of our release notes as part of our great ARMI release planning.

So... sure, I can remove it.  But expect to see it again soon!

Comment (john-science) [reply doc/release/0.2.rst]: Removed!

**Review by keckler — APPROVED** [review]
